ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/services/ClaimMigrator.java

```diff
@@ -100,14 +109,26 @@ private void migrateClaims(
         AtomicInteger failedOnUpdateMigrations
     ) {
         notMigratedClaims.parallelStream().forEach(claim -> {
-            migrationHandler.migrateClaim(
-                migratedClaims,
-                failedOnCreateMigrations,
-                failedOnUpdateMigrations,
-                updatedClaims,
-                claim,
-                user
-            );
+            if (fixDataIssues && isSettledOrJudgement(claim)) {
+                dataFixHandler.fixClaim(migratedClaims, failedOnUpdateMigrations, updatedClaims, claim, user);
+            } else {
```

Comment:
This condition have a leakage. This will try to migrate claim if the `isSettledOrJudgement(claim)` returns `false` which is wrong. 

We should not call migrateClaim at all if we have flag `fixDataIssues` as `true`.

Reply:
good one